### DISCUSSION TITLE
Crashes on Android (Google pixel 6a)

### DISCUSSION THREAD

**Discussion by Axcxiii** [ppy/osu#23214] [Q&A]
### Type

Crash to desktop

### Bug description

Tried to open latest version after installing Android 14 beta and just crashes. That's pretty much it.

### Version

2023.403.1

### Logs

 None, for some reason

**Comment by Art34222** [comment]
I have the same problem on my Pixel 7 with android 14 beta 1

**Reply by Susko3** [reply]
Did you say that you rolled back to Android 13 stable, and that you no longer have the issue? That is valuable information, so could be please confirm it.

**Reply by Art34222** [reply]
Yes, there is no problem on the 13 Android

**Reply by Art34222** [reply]
It's just not clear why I can't download osu on the emulator. Error with code 7 pops up into the logcat.

**Comment by Susko3** [comment]
Please provide logcat logs from when the crash happens:
1. Run `adb logcat -c` to clear the logs
2. Open the game, and let it crash
3. Run `adb logcat -d > logcat.txt` to save the log and upload it here

(If you're using the beta, I hope you know what adb and logcat are.)

**Reply by Axcxiii** [reply]
[logcat.txt](https://github.com/ppy/osu/files/11304984/logcat.txt)
I don't know if it was supposed to be this long.

**Reply by Susko3** [reply]
The crash is same as https://github.com/realm/realm-core/issues/4777#issuecomment-884043180. Looks like we'll need to put the realm database and other files into a different folder. We are currently using [`GetExternalFilesDir`](https://github.com/ppy/osu-framework/blob/a2bdbc9cbef46f6885e711dd447c28d479108bc0/osu.Framework.Android/AndroidGameHost.cs#L71), which is not the function described in the linked issue, and [should work](https://developer.android.com/reference/android/content/Context#getExternalFilesDir(java.lang.String)).
> Starting in `Build.VERSION_CODES.KITKAT`, no permissions are required to read or write to the returned path; it's always accessible to the calling app.

**Comment by Axcxiii** [comment]
Nevermind, I got it back to Android 13

**Reply by Art34222** [reply]
The bug is not fixed and you close the discussion?

**Comment by Art34222** [comment]
I think we can already close this thread, because this issue was reported in "Issues" (https://github.com/ppy/osu/issues/23773).
